src/Compilers/Core/MSBuildTask/Microsoft.Build.Tasks.CodeAnalysis.csproj

```diff
@@ -60,5 +60,45 @@
   <ItemGroup>
     <InternalsVisibleTo Include="Microsoft.Build.Tasks.CodeAnalysis.UnitTests" />
   </ItemGroup>
+
+  <!-- Generate a .targets file that sets $(CompilerApiVersion) to the current Roslyn version -->
+  <PropertyGroup>
+    <CompilerVersionTargetsFileIntermediatePath>$(IntermediateOutputPath)Microsoft.Managed.Core.CurrentVersions.targets</CompilerVersionTargetsFileIntermediatePath>
+  </PropertyGroup>
+
+  <Target Name="GenerateCompilerVersionTargets"
+          BeforeTargets="AssignTargetPaths"
+          DependsOnTargets="GenerateCompilerVersionTargetsFile">
+    <ItemGroup>
+      <Content Include="$(CompilerVersionTargetsFileIntermediatePath)">
+        <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+        <Pack>true</Pack>
+        <BuildAction>None</BuildAction>
+        <PackageCopyToOutput>true</PackageCopyToOutput>
+        <PackagePath>contentFiles\any\any</PackagePath>
+      </Content>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="GenerateCompilerVersionTargetsFile"
+          Inputs="$(MSBuildThisFileFullPath)"
+          Outputs="$(CompilerVersionTargetsFileIntermediatePath)">
+
+    <PropertyGroup>
+      <CompilerApiVersion>$([System.Version]::Parse($(VersionPrefix)).Major).$([System.Version]::Parse($(VersionPrefix)).Minor)</CompilerApiVersion>
```

Comment:
Sorry, I realize I should have brought this up sooner but shoudn't we prefix this property name with `Roslyn`? Basically `RoslynCompilerApiVersion` vs. `CompilerApiVersion`? Reasonable for other compilers to provide similar properties.

Reply:
> Reasonable for other compilers to provide similar properties.

The reason I didn't put `Roslyn` in property name is exactly for that reason - to support other compilers. The `CompilerApiVersion` version gets piped into the `ResolvePackageAssets` task in the dotnet/sdk:

https://github.com/dotnet/sdk/blob/500e1f61104f94753d7d84c3941d93fec2814291/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.PackageDependencyResolution.targets#L267-L272

```
    <ResolvePackageAssets
      ProjectAssetsFile="$(ProjectAssetsFile)"
      ProjectAssetsCacheFile="$(ProjectAssetsCacheFile)"
      ProjectPath="$(MSBuildProjectFullPath)"
      ProjectLanguage="$(Language)"
      CompilerApiVersion="$(CompilerApiVersion)"
```

I didn't want that task to have to understand compiler-specific properties (for example needing to read both `RoslynCompilerApiVersion` or `FSharpCompilerApiVersion`).

Instead, I just made a general `CompilerApiVersion` which can be set to `roslynX.Y` or `fsharpX.Y` or whatever. This way, the task shouldn't need to be updated to support other compilers in the future.